### PR TITLE
Extract plugin to its own repo

### PR DESCRIPTION
**Pull request by tarcisiozf**

### CODE REVIEW

File: lib/plugin.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    this._scheduleAutoRenewal(expiresAt)
+  }
+
+  * _getManagers () {
+    for (const [id, ref] of this._algoManagers.entries()) {
+      const manager = ref.deref()
+
+      if (!manager) {
+        this._algoManagers.delete(id)
```
Comment (ghost) [inline lib/plugin.js]: the method is named as getter, but it may change the state

File: lib/plugin.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    const { authToken, expiresAt } = response
+
+    for (const manager of this._getManagers()) {
+      manager.auth({ authToken })
+    }
+
+    this._scheduleAutoRenewal(expiresAt)
+  }
+
+  * _getManagers () {
```
Comment (ghost) [inline lib/plugin.js]: is it possible to get rid of the generator to simplify the code?

File: lib/adapters/bitfinex-adapter.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      transform: true
+    })
+  }
+
+  /**
+   * @returns {Promise<{authToken: string, expiresAt: number}>}
+   */
+  async refreshToken () {
+    const [authToken] = await this.rest.generateToken(this._generateOptions)
+    const expiresAt = Date.now() + (this._generateOptions.ttl * 1000)
```
Comment (vigan-abd) [inline lib/adapters/bitfinex-adapter.js:49]: maybe we could reduce it just a bit to avoid expire token error, e.g. (this._generateOptions.ttl * 0.8)

Comment (vigan-abd) [reply lib/adapters/bitfinex-adapter.js:49]: ok I see that we have THRESHOLD on plugin so please ignore my comment above :)

File: lib/plugin.js (hunk 2 above)
Comment (tarcisiozf) [reply lib/plugin.js]: 30ba35e73fc51594de07e3ddc2b4ab64ffbd18a8

File: lib/plugin.js (hunk 1 above)
Comment (tarcisiozf) [reply lib/plugin.js]: 30ba35e73fc51594de07e3ddc2b4ab64ffbd18a8

**Review by vigan-abd — APPROVED** [review]

**Pull request merged**